src/analyses/apron/apronPriv.apron.ml

```diff
@@ -573,7 +581,70 @@ struct
     include Printable.Std
     include SetDomain.Make (CilType.Varinfo)
   end
-  module LAD1 = MapDomain.MapBot (VS) (AD)
+  module LAD = MapDomain.MapBot (VS) (AD)
+
+  let keep_only_protected_globals ask m octs =
+    (* normal (strong) mapping: contains only still fully protected *)
+    let octs =
+      (* must filter by protection to avoid later meeting with non-protecting *)
+      LAD.filter (fun gs _ ->
+          VS.for_all (is_protected_by ask m) gs (* TODO: is this subset check right? *)
+        ) octs
+    in
+    LAD.map (keep_only_protected_globals ask m) octs (* TODO: is this even necessary if keys are filtered above? *)
```

Comment:
It should not be necessary here. Are there any tests that fail if we remove this?

Reply:
No, I removed it now. It was left around from all my experiments with also trying just non-empty intersections (which need to be filtered), but that happens separately anyway.